### PR TITLE
add Minestom implementation

### PR DESCRIPTION
**Pull request by HSGamer**
This PR add an Minestom module to support monitoring on Minestom servers.
Need further testing before merging.

### CODE REVIEW

**Comment by HSGamer** [comment]
Time to review

File: spark-common/src/main/java/me/lucko/spark/common/SparkPlatform.java [hunk 1]
```diff
@@ -142,7 +142,7 @@ public SparkPlatform(SparkPlugin plugin) {
 
         this.tickHook = plugin.createTickHook();
         this.tickReporter = plugin.createTickReporter();
-        this.tickStatistics = this.tickHook != null ? new TickStatistics() : null;
+        this.tickStatistics = this.tickHook != null || this.tickReporter != null ? new TickStatistics() : null;
```
Comment (lucko) [inline spark-common/src/main/java/me/lucko/spark/common/SparkPlatform.java:145]: The assumption in spark currently is that if there is a `tickReporter` then there is also a `tickHook`.  I think Minestom should be able to inplement both with the events that are available?

File: spark-minestom/src/main/java/me/lucko/spark/minestom/MinestomSparkExtension.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    }
+
+    @Override
+    public Path getPluginDirectory() {
+        return getDataDirectory();
+    }
+
+    @Override
+    public String getCommandName() {
+        return "sparkms";
```
Comment (lucko) [inline spark-minestom/src/main/java/me/lucko/spark/minestom/MinestomSparkExtension.java:57]: I think just `"spark"` will be ok here? :)

File: spark-minestom/src/main/java/me/lucko/spark/minestom/MinestomSparkExtension.java [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    public void terminate() {
+        this.platform.disable();
+        MinecraftServer.getCommandManager().unregister(command);
+    }
+
+    private static final class MinestomSparkCommand extends Command {
+        public MinestomSparkCommand(MinestomSparkExtension extension) {
+            super("sparkms");
+            setDefaultExecutor((sender, context) -> extension.platform.executeCommand(new MinestomCommandSender(sender), new String[0]));
+            var arrayArgument = ArgumentType.StringArray("query");
```
Comment (lucko) [inline spark-minestom/src/main/java/me/lucko/spark/minestom/MinestomSparkExtension.java:124]: Just a codestyle thing, please could you replace the `var`s in this method with actual types. Makes it much easier to read the code without an IDE.

File: spark-minestom/src/main/java/me/lucko/spark/minestom/MinestomTickReporter.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+import me.lucko.spark.common.tick.AbstractTickReporter;
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.event.Event;
+import net.minestom.server.event.EventNode;
+import net.minestom.server.event.server.ServerTickMonitorEvent;
+
+import java.util.UUID;
+
+public class MinestomTickReporter extends AbstractTickReporter {
+    private final EventNode<Event> node = EventNode.all("sparkTickReporter-" + UUID.randomUUID());
```
Comment (lucko) [inline spark-minestom/src/main/java/me/lucko/spark/minestom/MinestomTickReporter.java:32]: I know nothing about Minestom, but is this the correct way to implement this listener?

`EventNode.all(...)` ??

**Review by lucko — COMMENTED** [review]
Looks good - thanks for this! :)

Just a few comments for your consideration

**Comment by HSGamer** [comment]
After merging, you can add `minestom-extension` to your topics to make it appear at [minestom.net](minestom.net)

**Comment by lucko** [comment]
Thanks for the PR! :)

**Pull request merged**